sympy/printing/tests/test_glsl.py

```diff
@@ -381,24 +383,46 @@ def test_Matrix_printing():
     m = Matrix([[sin(q[1,0]), 0, cos(q[2,0])],
         [q[1,0] + q[2,0], q[3, 0], 5],
         [2*q[4, 0]/q[1,0], sqrt(q[0,0]) + 4, 0]])
-    assert glsl_code(m,M) == (
-'''M[0][0] = sin(q[1]);
+    assert glsl_code(m,M) == textwrap.dedent('''\
+M[0][0] = sin(q[1]);
 M[0][1] = 0;
 M[0][2] = cos(q[2]);
 M[1][0] = q[1] + q[2];
 M[1][1] = q[3];
 M[1][2] = 5;
 M[2][0] = 2*q[4]/q[1];
 M[2][1] = sqrt(q[0]) + 4;
-M[2][2] = 0;'''
-        )
+M[2][2] = 0;''')
 
 def test_Matrices_1x7():
     gl = glsl_code
     A = Matrix([1,2,3,4,5,6,7])
     assert gl(A) == 'float[7](1, 2, 3, 4, 5, 6, 7)'
     assert gl(A.transpose()) == 'float[7](1, 2, 3, 4, 5, 6, 7)'
 
+def test_Matrices_1x7_array_type_int():
+    gl = glsl_code
+    A = Matrix([1,2,3,4,5,6,7])
+    assert gl(A, array_type='int') == 'int[7](1, 2, 3, 4, 5, 6, 7)'
+
+def test_Tuple_array_type_custom():
+    gl = glsl_code
+    A = symbols('a b c')
+    assert gl(A, array_type='AbcType', glsl_types=False) == 'AbcType[3](a, b, c)'
+
+def test_Matrices_1x7_assign_to_symbols():
+    gl = glsl_code
+    A = Matrix([1,2,3,4,5,6,7])
+    assign_to = symbols('x.a x.b x.c x.d x.e x.f x.g')
+    assert gl(A, assign_to=assign_to) == textwrap.dedent('''\
+x.a = 1;
+x.b = 2;
+x.c = 3;
+x.d = 4;
+x.e = 5;
+x.f = 6;
+x.g = 7;''')
```

Comment:
The whole point of my `textwrap.dedent` suggestion was to allow you to indent these lines!